platform/reworkd_platform/services/worker/execution_engine.py

```diff
@@ -90,24 +93,38 @@ def create_execution_plan(
             ),
         )
 
-    def get_pruned_queue(self, if_condition: IfCondition, branch: bool) -> List[Node]:
+    def bfs_keep(self, node_ids_to_keep: List[str]) -> List[Node]:
+        visited = set(node_ids_to_keep)  # Initialize visited set with nodes to keep
+        queue = deque(node_ids_to_keep)  # Use a deque for efficient popleft operation
```

Comment:
Instead of leaving inline docs here. Might be good too leave a quick doc on the function eg:
```suggestion
    def bfs_keep(self, node_ids_to_keep: List[str]) -> List[Node]:
        """ explain algo """
        visited = set(node_ids_to_keep)
        queue = deque(node_ids_to_keep)
```

Reply:
Going to just remove comments actually. BFS is pretty standard